ipn/store/stores.go

```diff
@@ -173,16 +173,22 @@ func (s *FileStore) ReadState(id ipn.StateKey) ([]byte, error) {
 }
 
 // WriteState implements the StateStore interface.
-func (s *FileStore) WriteState(id ipn.StateKey, bs []byte) error {
+func (s *FileStore) WriteState(id ipn.StateKey, bs []byte) (err error) {
```

Comment:
naming results just for defer but then uglifying public godoc is one of my least loved parts of Go.

Normally I'd break Foo into Foo+foo and only used named results on the unexported foo, but in this case I don't think we need defer? don't update the map until the end and put an err != nil return err check on the WriteFile?